### PR TITLE
Add common vector constants to JS, add some missing Quat functionality

### PR DESCRIPTION
**Pull request by jherico**
Second attempt since my first broke scripts owing to the important distinction between glm::vec3 and vec3 when registering metatypes.

### CODE REVIEW

**Comment by ZappoMan** [comment]
Let's inline these one liners.

**Comment by jherico** [comment]
What's the point of inlining a function that's only going to get called from Javascript?  At best it will break the scripts again.

**Comment by jherico** [comment]
test this please

**Comment by hifi-gustavo** [comment]
Build console output for commit: e80b514f5804e52da244507444fc7d875542ca6d available at: https://gist.github.com/2a18e1d145d81e10917d<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5682.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5682.dmg

**Comment by murillodigital** [comment]
test this please

**Comment by hifi-gustavo** [comment]
Build console output for commit: e80b514f5804e52da244507444fc7d875542ca6d available at: https://gist.github.com/2a18e1d145d81e10917d<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5682.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5682.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 75bfacd0c057d4e5b444939c447c270356b83e13 available at: https://gist.github.com/2a18e1d145d81e10917d<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5682.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5682.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: bc94346c29722a2a0d6470a54a2e7f14dda3a458 available at: https://gist.github.com/2a18e1d145d81e10917d<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5682.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5682.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 69cf63f7240e5ae4aaa9c1082da5f46a610ada59 available at: https://gist.github.com/9faa3d13a75e1e213028<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5682.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5682.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: be594d81f973383b134e4174a336eca262ca6ac7 available at: https://gist.github.com/2a18e1d145d81e10917d<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5682.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5682.dmg

File: libraries/script-engine/src/Vec3.cpp [hunk 1]
```diff
@@ -123,3 +125,70 @@ glm::vec3 Vec3::fromPolar(float elevation, float azimuth) {
     return fromPolar(v);
 }
 
+const vec3& Vec3::UNIT_X() {
```
Comment (ZappoMan) [inline libraries/script-engine/src/Vec3.cpp]: for consistency sake these should be: "const glm::vec3&" - and I still think these should be inline in the header file.

**Comment by hifi-gustavo** [comment]
Build console output for commit: effee1f6fa7eee2eb42c5463b9c68691dd1de221 available at: https://gist.github.com/2a18e1d145d81e10917d<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5682.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5682.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 1ab449cb52c1fe8d3c7e4bd7a9ce49b74c99c06c available at: https://gist.github.com/2a18e1d145d81e10917d<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5682.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5682.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 8dc390576710f11d59b9ba6e9d5da70c2da8b73f available at: https://gist.github.com/2a18e1d145d81e10917d<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5682.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5682.dmg

**Pull request merged**